plugins/modules/zos_operator.py

```diff
@@ -130,38 +170,161 @@ def run_module():
 
 
 def parse_params(params):
     arg_defs = dict(
         cmd=dict(arg_type="str", required=True),
         verbose=dict(arg_type="bool", required=False),
-        debug=dict(arg_type="bool", required=False),
+        wait_time_s=dict(arg_type="int", required=False),
+        wait=dict(arg_type="bool", required=False),
     )
     parser = BetterArgParser(arg_defs)
     new_params = parser.parse_args(params)
     return new_params
 
 
 def run_operator_command(params):
+    # Usage: (rexfile) wait_time_s command [-v] [-n]
+    #       -v: print out verbose security information
+    #       -n: nowait
+
+    script = """/*rexx*/
+wait_time = __argv.2
+command = __argv.3
+verbosemode = __argv.4
+nowait = __argv.5
+
+Address 'TSO'
+IsfRC = isfcalls( "ON" )
+showoutput = 0
+waitmsg = " WAIT"
+if nowait == "NOWAIT" then do
+  waitmsg = ""
+  end
+
+verbose = ""
+if verbosemode == "VERB" then do
+  showoutput = 1
+  ISFSECTRACE="ON"
+  verbose = "( VERBOSE" waitmsg ")"
+  end
+else do
+  if waitmsg == " WAIT" then do
+    verbose = "( WAIT )"
+    end
+  end
+
+if wait_time > 0 then do
+  ISFDELAY=wait_time
+  end
+
+sdsfcmd = False
+
+fwd = WORD(command, 1)
+ffwd = translate(fwd)
+fch = SUBSTR( fwd, 1, 1)
+if( ffwd == 'QUERY' | ffwd == 'SET' | ffwd == 'WHO' | fch == '/' ) then
+  do
+    sdsfcmd = True
+  end
+if sdsfcmd == True then
+  do
+    address SDSF "ISFEXEC " command verbose
+  end
+else
+  do
+    address SDSF "ISFEXEC '/"command"'" verbose
+  end
+saverc = rc
+
+IsfRC = isfcalls( "OFF" )
+trace Off
+if isfulog.0 > 0 then
+  do
+    do ix=1 to isfulog.0
+      say isfulog.ix
+    end
+  end
+if isfresp.0 > 0 then
+  do
+    do ix=1 to isfresp.0
+      say isfresp.ix
+    end
+  end
+if showoutput > 0 then
+  do
+    SAY "===================="
+    SAY "result code: " saverc
+    SAY "===================="
+
+    say ""
+    say "Action messages"
+    say isfmsg
+    say ""
+    do ix=1 to isfmsg2.0
+      say isfmsg2.ix
+    end
+    say ""
+  end
+EXIT saverc
+"""
     module = AnsibleModuleHelper(argument_spec={})
-    command = params.get("cmd")
-    verbose = "-v" if params.get("verbose") else ""
-    debug = "-d" if params.get("debug") else ""
-    rc, stdout, stderr = module.run_command(
-        "opercmd {0} {1} {2}".format(verbose, debug, command),
-    )
-    message = stdout + stderr
+
+    fulline = " " + str(params.get("wait_time_s")) + " "
+
+    fulline += '"' + params.get("cmd") + '"'
+
+    if params.get("verbose"):
+        fulline += " VERB"
+    else:
+        fulline += " QUIET"
+
+    if params.get("wait"):
+        fulline += " WAIT"
+    else:
+        fulline += " NOWAIT"
+
+    delete_on_close = True
+    tmp_file = NamedTemporaryFile(delete=delete_on_close)
+    with open(tmp_file.name, "w") as f:
+        f.write(script)
+    chmod(tmp_file.name, S_IEXEC | S_IREAD | S_IWRITE)
+
+    rc, stdout, stderr = module.run_command(tmp_file.name + fulline)
+
+    message = stdout + stderr + "\n" + "Ran " + fulline + "\n"
```

Comment:
I think you have an edge case bug here

In the event that stdout and stderr are both empty strings, the `split("\n")` in `run_module()` will separate the message string as follows
```
[ "", "Ran " + fulltime, ""]
```
Similarly, if stdout and stderr are both strings but do not contain a newline character:
```
[stdout + stderr, "Ran " + fulltime, ""]
```
In `run_module()`, you are expecting the response line to always be at index 2, which in these cases would be an empty string, causing issues in the remainder of the module. 

In addition, if your stdout and stderr strings contain multiple newlines, index 2 would contain part of the content of stdout or stderr, not the response line you are expecting.

My recommendation:

- Add additional keys to your response in `run_operator_command()` to account for the stdout, stderr, and fulline fields instead of combining them and then trying to parse them in `run_module()`, this way, you are guaranteed to always be accessing the correct data.

Reply:
While I didn't think stdout or stderr could return without a newline (except for None), I definitely see the potential... good eyes!  My prior code was just checking to make sure there were > 3 lines.  I'll get a bit more elegant on this today.